runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -106,6 +106,13 @@ public AddressSpaceView(@Nonnull final CorfuRuntime runtime) {
         cacheBuilder.maximumSize(maxCacheEntries);
         log.info("AddressSpaceView readCache size is set to {}", maxCacheEntries);
 
+        if (maxCacheEntries == 0) {
+            log.warn("Since AddressSpaceView readCache size is 0, " +
+                    "overriding CorfuRuntime bulkReadSize and checkpointReadBatchSize to 1.");
+            runtime.getParameters().setBulkReadSize(1);
+            runtime.getParameters().setCheckpointReadBatchSize(1);
+        }
+
```

Comment:
If the batch can be successfully read and later GC'd/dropped since it's not cached, then that implies that there is enough memory to keep that batch in the cache.

Why not set the cache size to the batch size for the compactor? that could improve the compactors performance.

Reply:
Done.